tests/platform_tests/daemon/test_syseepromd.py

```diff
@@ -0,0 +1,209 @@
+"""
+Check daemon status inside PMON container. Each daemon status is checked under the conditions below in this script:
+* Daemon Running Status 
+* Daemon Stop status
+* Daemon Restart status
+
+This script is to cover the test case in the SONiC platform daemon and service test plan:
+https://github.com/Azure/sonic-mgmt/blob/master/docs/testplan/PMON-Services-Daemons-test-plan.md
+"""
+import logging
+import re
+import time
+
+from datetime import datetime
+
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.platform.daemon_utils import check_pmon_daemon_enable_status
+from tests.common.platform.processes_utils import wait_critical_processes, check_critical_processes
+from tests.common.utilities import compose_dict_from_cli
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any'),
+    pytest.mark.sanity_check(skip_sanity=True),
+    pytest.mark.disable_loganalyzer
+]
+
+expected_running_status = "RUNNING"
+expected_stopped_status = "STOPPED"
+expected_exited_status = "EXITED"
+
+daemon_name = "syseepromd"
+
+SIG_STOP_SERVICE = None
+SIG_TERM = "-15"
+SIG_KILL = "-9"
+
+syseepromd_tbl_key = ""
+
+@pytest.fixture(scope="module", autouse=True)
+def setup(duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+    daemon_en_status = check_pmon_daemon_enable_status(duthost, daemon_name)
+    if daemon_en_status is False:
+        pytest.skip("{} is not enabled in {}".format(daemon_name, duthost.facts['platform'], duthost.os_version))
+
+
+@pytest.fixture(scope="module", autouse=True)
+def teardown_module(duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+    yield
+
+    daemon_status, daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    if daemon_status is not "RUNNING":
+        duthost.start_pmon_daemon(daemon_name)
+        time.sleep(10)
+    logger.info("Tearing down: to make sure all the critical services, interfaces and transceivers are good")
+    check_critical_processes(duthost, watch_secs=10)
+
+
+@pytest.fixture
+def check_daemon_status(duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+    daemon_status, daemon_pid = duthost.get_pmon_daemon_status(daemon_name)
+    if daemon_status is not "RUNNING":
+        duthost.start_pmon_daemon(daemon_name)
+        time.sleep(10)
+
+def collect_data(duthost):
+    keys = duthost.shell('redis-cli -n 6 keys "EEPROM_INFO|*"')['stdout_lines']
+
+    dev_data = {}
+    for k in keys:
+        data = duthost.shell('redis-cli -n 6 hgetall "{}"'.format(k))['stdout_lines']
```

Comment:
suggest to use **sonic-db-cli** instead of 'redis-cli' as **sonic-db-cli** is more generic and less likely to change across versions